### DISCUSSION TITLE
Getting error to call TestBed.initTestEnvironment() first on all tests

### DISCUSSION THREAD

**Discussion by srycroft** [orta/vscode-jest#1026] [Q&A]
I can run all tests fine via command line `npx ng test`.

Running any single test I get an error like this

```
  ● SchoolsEffects › given that a Get Schools action is dispatched › should fetch the schools based on the filters

    Need to call TestBed.initTestEnvironment() first

      81 |     routerStub = { navigate: jest.fn() };
      82 |
    > 83 |     await TestBed.configureTestingModule({
         |                   ^
      84 |       providers: [
      85 |         SchoolsEffects,
      86 |         provideMockActions(() => actions$),
```

and then later

```
  ● SchoolsEffects › given that a Get Schools action is dispatched › should fetch the schools based on the filters

    zone-testing.js is needed for the fakeAsync() test helper but could not be found.
            Please make sure that your environment includes zone.js/testing

      at resetFakeAsyncZone (node_modules/@angular/core/fesm2020/testing.mjs:252:11)
      at Object.<anonymous> (node_modules/@angular/core/fesm2020/testing.mjs:24510:13)
```

jest.config.js contents:
```
module.exports = {
  preset: 'jest-preset-angular',
  globalSetup: 'jest-preset-angular/global-setup',
  testMatch: [
    '**/__tests__/**/*.[jt]s?(x)',
    '**/?(*.)+(spec|test|pact).[jt]s?(x)'
  ]
};
```

Again, the tests run fine via npx ng test, but running via the extension always fails. I haven't configured any extension settings from the default.

Any ideas?

**Comment by connectdotz** [comment]
If you have a "test" in package.json, you can set `"jest.jestCommandLine": "yarn test"` or `"jest.jestCommandLine": "npm run test --"`. Or you can try npx directly `"jest.jestCommandLine": "npx ng test"`

**Reply by srycroft** [reply]
Now getting an error when initializing
```
Error: Unknown arguments: listTests, watchAll


[error] failed to retrieve test file list. TestExplorer might show incomplete test items
```
And I can't even attempt to run tests - the buttons don't show.

**Reply by connectdotz** [reply]
please share the full command line and your jest.jestCommandLine setting...

**Reply by srycroft** [reply]
Thought a screenshot would help. The initial 'auto-config' output happened before I set `jest.jestCommandLine` I believe.

<img width="734" alt="jestVsCode" src="https://github.com/jest-community/vscode-jest/assets/4493787/98587b63-7d81-4e9b-804f-c62f09f773fc">

**Comment by shamoon** [comment]
Im having a similar issue. Hopefully it's not 'hijacking' your thread OP. The fix feels a little hack-y so if anyone has advice I appreciate it but here's how I solved it.

First, adding 
```js
import 'jest-preset-angular/setup-jest'
```
to the top of my `setup-jest.ts` file seems to make tests work via vscode-jest

However when I do that tests fail when running just plain `npm run test` (and thus will fail on my ci) with errors like:
```bash
  ● Test suite failed to run

    Cannot set base providers because it has already been called

      1 | import { jest } from '@jest/globals'
    > 2 | import 'jest-preset-angular/setup-jest';
```

One difference from you OP is I dont have `globalSetup: 'jest-preset-angular/global-setup',` in my `jest.config.js`

See https://stackoverflow.com/questions/69882564/jest-error-cannot-set-base-providers-after-angular-v13-upgrade-using-ng-test/73825433#73825433 but solution there left me in the same place, tests either work using vscode-jest or CLI but not both...

So in the end I put condition around the above import, e.g. in `setup-jest.ts`:
```js
if (process.env.NODE_ENV === 'test') {
  require('jest-preset-angular/setup-jest')
}
```

and things work in both places:
```bash
Test Suites: 26 passed, 26 total
Tests:       140 passed, 140 total
Snapshots:   0 total
Time:        4.678 s
Ran all test suites.
```

including those glorious green checks:
<img width="84" alt="Screenshot 2023-05-26 at 11 13 37 AM" src="https://github.com/jest-community/vscode-jest/assets/4887959/319cf693-8747-47b6-a325-939c134e85bd">

TBH, find this stuff pretty confusing, appreciate any help!
